File: 1.0-computer_science/Algorithms/0.0-python/0.0-tow_Sum.py
# Given an array of intigers, return indices of the two numbers such that they add up
# to a specific target.
# You may assume that ech input would have  exaclty one solution.

class Solution:

    def find_Index(self, arr, target):
        """
        :type arr: List[int]
        :type target: int 
        """
        num_to_index = {}

        for x, y  in enumerate(arr):
            print(target - y)
            if targe - y in num_to_index:
                return [num_to_index[target - y], x]

            num_to_index[y] = x

        return []


targe = 15 
num = [1, 2, 5, 6, 7, 10, 22]
objetive = Solution()
print(objetive.find_Index(num, targe))

File: 1.0-computer_science/Algorithms/1.0-javascript/0.0-tow_Sum.js
// # Given an array of intigers, return indices of the two numbers such that they add up
// # to a specific target.
// # You may assume that ech input would have  exaclty one solution.
class Solution{

    indIndex(arr, target){


        for (let i = 0; i < arr.length; i++){
            if(arr.includes(target - arr[i])){
                const idx2 = arr.indexOf(target - arr[i])
                return [i, idx2]
            }
        }

        return []
    }
}


const arr = [1,2,3,5,41,8,10,9,11]
const target = 100
const newOject = new Solution
console.log(newOject.indIndex(arr, target))
